src/models/installation.ts

```diff
@@ -127,7 +147,43 @@ Installation.init({
 		type: DataTypes.INTEGER,
 		allowNull: true
 	}
-}, { sequelize });
+}, {
+	hooks: {
+		beforeSave: async (instance: Installation, opts) => {
+			if (opts.fields?.includes("sharedSecret")) {
+				if (!instance.encryptedSharedSecret) {
+					//if caller only set the sharedSecret, copy it to encryptedSharedSecret;
+					instance.encryptedSharedSecret = instance.sharedSecret;
```

Comment:
`await instance.encryptChangedSecretFields(opts.fields);`, but the field that changed could simply just be `sharedSecret`, which we're saving to `encryptedSharedSecret`.  However, the `encryptChangedSecretFields` is looking for which fields has changed and `encryptedSharedSecret` isn't one of them since the ORM only detected `sharedSecret` has changed.  To make sure that our encryption process is triggered, it need to see that `encryptedSharedSecret` is part of the `opts.fields` that has changed.

Hence, doing something like `opts.fields.push("encryptedSharedSecret")` if you've overwritten that `encryptedSharedSecret` with the value of `sharedSecret`

Reply:
GOOD PICKS!
Turns out i missed a test case. Have added it.